tt_metal/impl/program/program.cpp

```diff
@@ -756,6 +756,9 @@ void Program::populate_dispatch_data(Device* device) {
                 sub_kernels = {kernel->processor()};
             }
 
+            // Kernels do not work on idle eth cores, see issue #7771.
```

Comment:
Hey @aliuTT  how does this placement look for assert? It was first place that came to mind in FD2 path that already have a branch for ETH cores, and fires in active-idle and inactive-idle eth core cases. Any concerns? (CI running).